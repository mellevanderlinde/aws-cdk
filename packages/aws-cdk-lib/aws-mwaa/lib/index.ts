export * from './mwaa.generated';
export * from './environment';