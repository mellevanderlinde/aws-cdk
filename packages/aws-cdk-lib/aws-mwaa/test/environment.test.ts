import { Template } from '../../assertions';
import * as ec2 from '../../aws-ec2/lib';
import * as iam from '../../aws-iam/lib';
import * as kms from '../../aws-kms/lib';
import * as logs from '../../aws-logs';
import * as s3 from '../../aws-s3/lib';
import * as cdk from '../../core';
import * as mwaa from '../lib';

describe('Environment', () => {
  describe('environment tests', () => {

    let stack: cdk.Stack;
    let bucket: s3.Bucket;
    let dagS3Path: string;
    let vpc: ec2.Vpc;
    let name: string;
    let securityGroups: ec2.SecurityGroup[];
    let subnet1: ec2.Subnet;
    let subnet2: ec2.Subnet;
    let subnets: ec2.Subnet[];

    beforeEach(() => {
      stack = new cdk.Stack();
      bucket = new s3.Bucket(stack, 'Bucket');
      dagS3Path = 'dags';
      vpc = new ec2.Vpc(stack, 'Vpc');
      name = 'Airflow';
      securityGroups = [new ec2.SecurityGroup(stack, 'SecurityGroup', { vpc })];
      subnet1 = new ec2.Subnet(stack, 'subnet1', {
        vpcId: vpc.vpcId,
        availabilityZone: vpc.availabilityZones[0],
        cidrBlock: vpc.vpcCidrBlock,
      });
      subnet2 = new ec2.Subnet(stack, 'subnet2', {
        vpcId: vpc.vpcId,
        availabilityZone: vpc.availabilityZones[1],
        cidrBlock: vpc.vpcCidrBlock,
      });
      subnets = [subnet1, subnet2];
    });

    test('all defaults', () => {
      const environment = new mwaa.Environment(stack, 'Environment', {
        bucket,
        dagS3Path,
        name,
        securityGroups,
        subnets,
      });

      const template = Template.fromStack(stack);
      template.resourceCountIs('AWS::MWAA::Environment', 1);
      expect(template).toMatchSnapshot();

      expect(environment.airflowVersion).toBe('2.8.1');
      expect(environment.environmentClass).toBe('mw1.small');
      expect(environment.environmentArn).toBeDefined();
      expect(environment.environmentName).toBe('Airflow');
      expect(environment.role).toBeInstanceOf(iam.Role);
    });

    test('execution role attached', () => {
      const role = new iam.Role(stack, 'ExecutionRole', {
        roleName: 'AttachedExecutionRole',
        assumedBy: new iam.CompositePrincipal(
          new iam.ServicePrincipal('airflow-env.amazonaws.com'),
          new iam.ServicePrincipal('airflow.amazonaws.com'),
        ),
      });

      new mwaa.Environment(stack, 'Environment', {
        bucket,
        dagS3Path,
        name,
        role,
        securityGroups,
        subnets,
      });

      const template = Template.fromStack(stack);

      template.resourceCountIs('AWS::MWAA::Environment', 1);
      template.hasResourceProperties('AWS::MWAA::Environment', {
        ExecutionRoleArn: { 'Fn::GetAtt': ['ExecutionRole605A040B', 'Arn'] },
      });

      template.resourceCountIs('AWS::IAM::Role', 1);
      template.hasResourceProperties('AWS::IAM::Role', {
        RoleName: 'AttachedExecutionRole',
        AssumeRolePolicyDocument: {
          Statement: [
            {
              Action: 'sts:AssumeRole',
              Effect: 'Allow',
              Principal: {
                Service: 'airflow-env.amazonaws.com',
              },
            }, {
              Action: 'sts:AssumeRole',
              Effect: 'Allow',
              Principal: {
                Service: 'airflow.amazonaws.com',
              },
            },
          ],
        },
      });
    });

    test('incorrect number of security groups', () => {
      expect(() => new mwaa.Environment(stack, 'Environment1', {
        bucket,
        dagS3Path,
        name,
        securityGroups: [],
        subnets,
      })).toThrow('Received 0 security groups, while between 1 and 5 are required');

      expect(() => new mwaa.Environment(stack, 'Environment2', {
        bucket,
        dagS3Path,
        name,
        securityGroups: [
          new ec2.SecurityGroup(stack, 'SecurityGroup1', { vpc }),
          new ec2.SecurityGroup(stack, 'SecurityGroup2', { vpc }),
          new ec2.SecurityGroup(stack, 'SecurityGroup3', { vpc }),
          new ec2.SecurityGroup(stack, 'SecurityGroup4', { vpc }),
          new ec2.SecurityGroup(stack, 'SecurityGroup5', { vpc }),
          new ec2.SecurityGroup(stack, 'SecurityGroup6', { vpc }),
        ],
        subnets,
      })).toThrow('Received 6 security groups, while between 1 and 5 are required');
    });

    test('incorrect number of subnets', () => {
      expect(() => new mwaa.Environment(stack, 'Environment1', {
        bucket,
        dagS3Path,
        name,
        securityGroups,
        subnets: [],
      })).toThrow('Received 0 subnet(s), while 2 are required');

      expect(() => new mwaa.Environment(stack, 'Environment2', {
        bucket,
        dagS3Path,
        name,
        securityGroups,
        subnets: [subnet1],
      })).toThrow('Received 1 subnet(s), while 2 are required');

      expect(() => new mwaa.Environment(stack, 'Environment3', {
        bucket,
        dagS3Path,
        name,
        securityGroups,
        subnets: [
          subnet1, subnet2,
          new ec2.Subnet(stack, 'subnet3', {
            vpcId: vpc.vpcId,
            availabilityZone: vpc.availabilityZones[0],
            cidrBlock: vpc.vpcCidrBlock,
          }),
        ],
      })).toThrow('Received 3 subnet(s), while 2 are required');
    });

    test('schedulers specified', () => {
      expect(() => new mwaa.Environment(stack, 'Environment2', {
        bucket,
        dagS3Path,
        name,
        securityGroups,
        schedulers: 1,
        subnets,
      })).toThrow('Number of specified schedulers is 1, while it must be between 2 to 5.');

      expect(() => new mwaa.Environment(stack, 'Environment3', {
        bucket,
        dagS3Path,
        name,
        securityGroups,
        schedulers: 6,
        subnets,
      })).toThrow('Number of specified schedulers is 6, while it must be between 2 to 5.');
    });

    test('optional properties specified', () => {
      const environment = new mwaa.Environment(stack, 'Environment', {
        accessMode: mwaa.AccessMode.PRIVATE_ONLY,
        airflowConfigurations: { 'core.default_timezone': 'utc' },
        airflowVersion: mwaa.AirflowVersion.V2_7_2,
        bucket,
        dagS3Path,
        endpointManagement: mwaa.EndpointManagement.SERVICE,
        environmentClass: mwaa.EnvironmentClass.MW1_LARGE,
        kmsKey: new kms.Key(stack, 'Key'),
        maxWorkers: 5,
        minWorkers: 2,
        name,
        pluginsVersion: 'plugins-hash',
        requirementsS3Path: 'requirements-path',
        requirementsVersion: 'requirements-hash',
        securityGroups,
        subnets,
        startupScriptS3Path: 'startup-path',
        startupScriptVersion: 'startup-version',
        tags: { key: 'value' },
        weeklyMaintenanceWindowStart: 'TUE:03:30',
      });

      const template = Template.fromStack(stack);
      expect(template).toMatchSnapshot();

      expect(environment.airflowVersion).toBe('2.7.2');
      expect(environment.environmentClass).toBe('mw1.large');
    });

    test('log groups specified', () => {
      const environment = new mwaa.Environment(stack, 'Environment', {
        bucket,
        dagS3Path,
        name,
        securityGroups,
        subnets,
        dagProcessingLogGroup: new logs.LogGroup(stack, 'DagLogs'),
        webserverLogGroup: new logs.LogGroup(stack, 'WebserverLogs'),
        logLevel: mwaa.LogLevel.WARNING,
      });
    });
  });
});
