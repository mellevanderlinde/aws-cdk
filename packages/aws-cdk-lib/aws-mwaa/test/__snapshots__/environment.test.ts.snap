// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Environment environment tests all defaults 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "Bucket83908E77": {
      "DeletionPolicy": "Retain",
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "Environment78414C64": {
      "Properties": {
        "AirflowVersion": "2.8.1",
        "DagS3Path": "dags",
        "EnvironmentClass": "mw1.small",
        "ExecutionRoleArn": {
          "Fn::GetAtt": [
            "EnvironmentExecutionRoleA0E74382",
            "Arn",
          ],
        },
        "MaxWorkers": 1,
        "MinWorkers": 1,
        "Name": "Airflow",
        "NetworkConfiguration": {
          "SecurityGroupIds": [
            {
              "Fn::GetAtt": [
                "SecurityGroupDD263621",
                "GroupId",
              ],
            },
          ],
          "SubnetIds": [
            {
              "Ref": "subnet1Subnet16A4B3BD",
            },
            {
              "Ref": "subnet2SubnetF9569CD3",
            },
          ],
        },
        "Schedulers": 2,
        "SourceBucketArn": {
          "Fn::GetAtt": [
            "Bucket83908E77",
            "Arn",
          ],
        },
      },
      "Type": "AWS::MWAA::Environment",
    },
    "EnvironmentExecutionRoleA0E74382": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "airflow-env.amazonaws.com",
              },
            },
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "airflow.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "RoleName": "AmazonMWAA-Airflow-ExecutionRole",
      },
      "Type": "AWS::IAM::Role",
    },
    "EnvironmentExecutionRoleDefaultPolicy063C3CFA": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "airflow:PublishMetrics",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":airflow:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":environment/Airflow",
                  ],
                ],
              },
            },
            {
              "Action": "s3:ListAllMyBuckets",
              "Effect": "Deny",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "Bucket83908E77",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "Bucket83908E77",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "Bucket83908E77",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "Bucket83908E77",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "logs:CreateLogStream",
                "logs:CreateLogGroup",
                "logs:PutLogEvents",
                "logs:GetLogEvents",
                "logs:GetLogRecord",
                "logs:GetLogGroupFields",
                "logs:GetQueryResults",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":logs:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":log-group/airflow-Airflow-*",
                  ],
                ],
              },
            },
            {
              "Action": [
                "logs:DescribeLogGroups",
                "cloudwatch:PutMetricData",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": [
                "sqs:ChangeMessageVisibility",
                "sqs:DeleteMessage",
                "sqs:GetQueueAttributes",
                "sqs:GetQueueUrl",
                "sqs:ReceiveMessage",
                "sqs:SendMessage",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":sqs:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":*:airflow-celery-*",
                  ],
                ],
              },
            },
            {
              "Action": [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:GenerateDataKey*",
                "kms:Encrypt",
              ],
              "Condition": {
                "StringLike": {
                  "kms:ViaService": {
                    "Fn::Join": [
                      "",
                      [
                        "sqs.",
                        {
                          "Ref": "AWS::Region",
                        },
                        ".amazonaws.com",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "NotResource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":kms:*:",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":key/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "EnvironmentExecutionRoleDefaultPolicy063C3CFA",
        "Roles": [
          {
            "Ref": "EnvironmentExecutionRoleA0E74382",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "SecurityGroupDD263621": {
      "Properties": {
        "GroupDescription": "Default/SecurityGroup",
        "SecurityGroupEgress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound traffic by default",
            "IpProtocol": "-1",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "Vpc8378EB38": {
      "Properties": {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": [
          {
            "Key": "Name",
            "Value": "Default/Vpc",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "VpcIGWD7BA715C": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "Default/Vpc",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "VpcPrivateSubnet1DefaultRouteBE02A9ED": {
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": {
          "Ref": "VpcPublicSubnet1NATGateway4D7517AA",
        },
        "RouteTableId": {
          "Ref": "VpcPrivateSubnet1RouteTableB2C5B500",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcPrivateSubnet1RouteTableAssociation70C59FA6": {
      "Properties": {
        "RouteTableId": {
          "Ref": "VpcPrivateSubnet1RouteTableB2C5B500",
        },
        "SubnetId": {
          "Ref": "VpcPrivateSubnet1Subnet536B997A",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcPrivateSubnet1RouteTableB2C5B500": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "Default/Vpc/PrivateSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcPrivateSubnet1Subnet536B997A": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            0,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.128.0/18",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          {
            "Key": "Name",
            "Value": "Default/Vpc/PrivateSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcPrivateSubnet2DefaultRoute060D2087": {
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": {
          "Ref": "VpcPublicSubnet2NATGateway9182C01D",
        },
        "RouteTableId": {
          "Ref": "VpcPrivateSubnet2RouteTableA678073B",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcPrivateSubnet2RouteTableA678073B": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "Default/Vpc/PrivateSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcPrivateSubnet2RouteTableAssociationA89CAD56": {
      "Properties": {
        "RouteTableId": {
          "Ref": "VpcPrivateSubnet2RouteTableA678073B",
        },
        "SubnetId": {
          "Ref": "VpcPrivateSubnet2Subnet3788AAA1",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcPrivateSubnet2Subnet3788AAA1": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            1,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.192.0/18",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          {
            "Key": "Name",
            "Value": "Default/Vpc/PrivateSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcPublicSubnet1DefaultRoute3DA9E72A": {
      "DependsOn": [
        "VpcVPCGWBF912B6E",
      ],
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": {
          "Ref": "VpcIGWD7BA715C",
        },
        "RouteTableId": {
          "Ref": "VpcPublicSubnet1RouteTable6C95E38E",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcPublicSubnet1EIPD7E02669": {
      "Properties": {
        "Domain": "vpc",
        "Tags": [
          {
            "Key": "Name",
            "Value": "Default/Vpc/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "VpcPublicSubnet1NATGateway4D7517AA": {
      "DependsOn": [
        "VpcPublicSubnet1DefaultRoute3DA9E72A",
        "VpcPublicSubnet1RouteTableAssociation97140677",
      ],
      "Properties": {
        "AllocationId": {
          "Fn::GetAtt": [
            "VpcPublicSubnet1EIPD7E02669",
            "AllocationId",
          ],
        },
        "SubnetId": {
          "Ref": "VpcPublicSubnet1Subnet5C2D37C4",
        },
        "Tags": [
          {
            "Key": "Name",
            "Value": "Default/Vpc/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "VpcPublicSubnet1RouteTable6C95E38E": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "Default/Vpc/PublicSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcPublicSubnet1RouteTableAssociation97140677": {
      "Properties": {
        "RouteTableId": {
          "Ref": "VpcPublicSubnet1RouteTable6C95E38E",
        },
        "SubnetId": {
          "Ref": "VpcPublicSubnet1Subnet5C2D37C4",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcPublicSubnet1Subnet5C2D37C4": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            0,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.0.0/18",
        "MapPublicIpOnLaunch": true,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          {
            "Key": "Name",
            "Value": "Default/Vpc/PublicSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcPublicSubnet2DefaultRoute97F91067": {
      "DependsOn": [
        "VpcVPCGWBF912B6E",
      ],
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": {
          "Ref": "VpcIGWD7BA715C",
        },
        "RouteTableId": {
          "Ref": "VpcPublicSubnet2RouteTable94F7E489",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcPublicSubnet2EIP3C605A87": {
      "Properties": {
        "Domain": "vpc",
        "Tags": [
          {
            "Key": "Name",
            "Value": "Default/Vpc/PublicSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "VpcPublicSubnet2NATGateway9182C01D": {
      "DependsOn": [
        "VpcPublicSubnet2DefaultRoute97F91067",
        "VpcPublicSubnet2RouteTableAssociationDD5762D8",
      ],
      "Properties": {
        "AllocationId": {
          "Fn::GetAtt": [
            "VpcPublicSubnet2EIP3C605A87",
            "AllocationId",
          ],
        },
        "SubnetId": {
          "Ref": "VpcPublicSubnet2Subnet691E08A3",
        },
        "Tags": [
          {
            "Key": "Name",
            "Value": "Default/Vpc/PublicSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "VpcPublicSubnet2RouteTable94F7E489": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "Default/Vpc/PublicSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcPublicSubnet2RouteTableAssociationDD5762D8": {
      "Properties": {
        "RouteTableId": {
          "Ref": "VpcPublicSubnet2RouteTable94F7E489",
        },
        "SubnetId": {
          "Ref": "VpcPublicSubnet2Subnet691E08A3",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcPublicSubnet2Subnet691E08A3": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            1,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.64.0/18",
        "MapPublicIpOnLaunch": true,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          {
            "Key": "Name",
            "Value": "Default/Vpc/PublicSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcVPCGWBF912B6E": {
      "Properties": {
        "InternetGatewayId": {
          "Ref": "VpcIGWD7BA715C",
        },
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
    "subnet1RouteTable987042F4": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "Default/subnet1",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "subnet1RouteTableAssociationB59E8505": {
      "Properties": {
        "RouteTableId": {
          "Ref": "subnet1RouteTable987042F4",
        },
        "SubnetId": {
          "Ref": "subnet1Subnet16A4B3BD",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "subnet1Subnet16A4B3BD": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            0,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": {
          "Fn::GetAtt": [
            "Vpc8378EB38",
            "CidrBlock",
          ],
        },
        "Tags": [
          {
            "Key": "Name",
            "Value": "Default/subnet1",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "subnet2RouteTable88F9EAEF": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "Default/subnet2",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "subnet2RouteTableAssociation329C2685": {
      "Properties": {
        "RouteTableId": {
          "Ref": "subnet2RouteTable88F9EAEF",
        },
        "SubnetId": {
          "Ref": "subnet2SubnetF9569CD3",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "subnet2SubnetF9569CD3": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            1,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": {
          "Fn::GetAtt": [
            "Vpc8378EB38",
            "CidrBlock",
          ],
        },
        "Tags": [
          {
            "Key": "Name",
            "Value": "Default/subnet2",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
